Core/Src/main.c

```diff
@@ -212,6 +211,23 @@ int main(void)
           //HAL_TIM_PWM_ConfigChannel(&htim1, &sConfig, TIM_CHANNEL_1);
           HAL_TIM_PWM_Start(&htim1, TIM_CHANNEL_1);
           break;
+        case NODE_VIBRO:
+          uint32_t frequency;
+          uint32_t duration;
+          uint32_t duty;
+          memcpy(&frequency, &i2c_buffer[0], sizeof(frequency));
+          memcpy(&duration, &i2c_buffer[4], sizeof(duration));
+          memcpy(&duty, &i2c_buffer[8], sizeof(duty));
+          endTone = HAL_GetTick() + duration;
+
+          // TODO: make the prescaler precise and configurable
+          uint32_t val = (0xFFFF * 180) / frequency;
+          TIM1->ARR = val;
+          float dutyf = (float)duty / 100.0f;

```

Comment:
I would try to not use `float` since it would be the only on in the firmware and likely increases the binary size. Can you try to check if the `int` version is a bit smaller?

Reply:
done removing the float doesn't change the fw size